packages/design-mui/src/components/MuiFilter/MuiFilter.tsx

```diff
@@ -0,0 +1,181 @@
+
+  return (
+    <Grid container spacing={2} padding={"1.25rem"} width={"31.25rem"}>
+      <Grid item xs={4}>
+        <InputLabel id="select-column-label">Columns</InputLabel>
+        <Select
+          fullWidth
+          labelId="select-column-label"
+          id="simple-select"
+          value={column}
+          onChange={(ev) => setColumn(ev.target.value)}
+        >
+          {columnsList.map((item) => {
+            return (
+              <MenuItem key={item} value={item}>
+                {item}
+              </MenuItem>
+            )
+          })}
+        </Select>
+      </Grid>
+      <Grid item xs={4}>
+        <InputLabel id="select-operator-label">Operator</InputLabel>
+        <Select
+          fullWidth
+          labelId="select-operator-label"
+          id="simple-select"
+          value={operator}
+          onChange={(ev) => setOperator(ev.target.value)}
+        >
+          <MenuItem value="empty">empty</MenuItem>
+          <MenuItem value="$eq">equals</MenuItem>
+          <MenuItem value="ilike">contains</MenuItem>
+        </Select>
+      </Grid>
+      <Grid item xs={4}>
+        <InputLabel id="value-field-label">Value</InputLabel>
+        <TextField
+          placeholder="Filter Value"
+          id="value-field"
+          variant="outlined"
+          onChange={(ev) => setValue(ev.target.value)}
+          value={value}
+        />
+      </Grid>
+    </Grid>
+  )
+}
+
+const MuiFilterDialog: React.FC<DialogProps> = ({
+  attributes,
+  column,
+  open,
+  operator,
+  value,
+  setColumn,
+  setOpen,
+  setOperator,
+  setValue,
+  setFilter,
+}) => {
+  const clearFilter = () => {
+    setColumn("")
+    setOperator("")
+    setValue("")
+    setFilter(undefined)
+    setOpen(false)
+  }
+
+  const applyFilter = () => {
+    //On first pass, we are only supporting one filter row.
+    setFilter([
+      {
+        [column]: value,
+        operator: operator,
+      },
+    ])
+    setOpen(false)
+  }
+
+  return (
+    <Dialog open={open}>
+      <DialogContent>
+        <MuiFilterRow
+          column={column}
+          setColumn={setColumn}
+          operator={operator}
+          setOperator={setOperator}
+          attributes={attributes}
+          value={value}
+          setValue={setValue}
+        />
+      </DialogContent>
+      <DialogActions>
+        <Button onClick={() => setOpen(false)}>Cancel</Button>
+        <Button onClick={clearFilter}>Clear Filter</Button>
+        <Button onClick={applyFilter} disabled={column === ""}>
+          Apply Filter
+        </Button>
+      </DialogActions>
+    </Dialog>
+  )
+}
+
+export const MuiFilter: React.FC<XCollectionProps> = ({
+  allSchemas,
+  schemaName,
+  filter,
+  setFilter,
+}) => {
+  const [open, setOpen] = useState<boolean>(false)
+  const [operator, setOperator] = useState<string>("")
+  const [column, setColumn] = useState<string>("")
+  const [value, setValue] = useState<string>("")
```

Comment:
let's chat about this tomorrow but commenting here so we remember to talk about it 😄 
this might need to merged into 1 more complex state variable if we start to support multiple filters at once (useFilter->useFilters)

this is good for now though! 👍

Reply:
Yeah, I figured it was good enough for now :)